File: class-abstract-interface/src/classes/duck.ts
import { Animal } from "./animal";
import { Fly } from "../interfaces/fly";
import { Swim } from "../interfaces/swim";

export class Duck extends Animal implements Fly, Swim {
  constructor(name: string) {
    super(name);
  }

  makeSound(): void {
    console.log("Sonido de pato");
  }

  sleep(): void {
    console.log("El pato duerme");
  }

  fly(): void {
    console.log("Fly fly fly");
  }

  swim(): void {
    console.log("blub blub");
  }
}

File: clases_clasesAbstractas_Interfaz/interfaces/Duck.java
package interfaces;

public class Duck extends Animal implements Fly {
    private String color;

    public Duck(int age, String color) {
        super(age);
        this.color = color;
    }
    
    @Override
    public void makeNoise() {
        System.out.println("Quack Quack");
    }

    @Override
    public void fly() {
        System.out.println("El pato esta volando con " + amountOfWings + " alas.");
    }

}
